tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11287,6 +11287,22 @@ bool Sema::DiagnoseHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc) {
         Diags.Report(Loc, diag::err_hlsl_wg_nodetrackrwinputsharing_missing);
         return true;
       }
+    } else if (opCode == hlsl::IntrinsicOp::MOP_CalculateLevelOfDetail ||
+               opCode ==
+                   hlsl::IntrinsicOp::MOP_CalculateLevelOfDetailUnclamped) {
+      const auto *shaderModel =
+          hlsl::ShaderModel::GetByName(getLangOpts().HLSLProfile.c_str());
```

Comment:
It's overkill for this change, but there are so many places in SemHLSL where we retrieve the profile by a string that we might as well encode at least the stage and the version numbers into the LangOpts directly